### PR TITLE
wasm4 2.5.3 (new formula)

### PR DESCRIPTION
**Pull request by FnControlOption**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  wasm4:
    * Binaries built for a non-native architecture were installed into wasm4's prefix.
      The offending files are:
        /opt/homebrew/Cellar/wasm4/2.5.3/libexec/lib/node_modules/wasm4/assets/natives/wasm4-mac	(x86_64)
```

**Review by chenrui333 — DISMISSED** [review]

File: Formula/wasm4.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  def install
+    cd "runtimes/native" do
+      system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+      system "cmake", "--build", "build"
+      if OS.mac?
+        (buildpath/"cli/assets/natives").install "build/wasm4" => "wasm4-mac"
+      else
+        (buildpath/"cli/assets/natives").install "build/wasm4" => "wasm4-linux"
+      end
```
Comment (SMillerDev) [inline Formula/wasm4.rb:33]: no cmake `--install`?

File: Formula/wasm4.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      system npm, "install"
+      system npm, "run", "build"
+      system bin/"w4", "bundle", "--html", "hello.html", "build/cart.wasm"
+      assert_predicate Pathname.pwd/"hello.html", :exist?
+      if OS.mac?
+        system bin/"w4", "bundle", "--mac", "hello", "build/cart.wasm"
+      else
+        system bin/"w4", "bundle", "--linux", "hello", "build/cart.wasm"
+      end
+      assert_predicate Pathname.pwd/"hello", :exist?
```
Comment (SMillerDev) [inline Formula/wasm4.rb:63]: Can we check that it runs?

**Comment by chenrui333** [comment]
ping @FnControlOption

**Comment by chenrui333** [comment]
ping @FnControlOption

**Comment by chenrui333** [comment]
@FnControlOption, can you do a rebase for the new bottle generation in this PR? Thanks!

File: Formula/wasm4.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/wasm4.rb:33]: Why would it need to be renamed?

**Review by chenrui333 — DISMISSED** [review]

File: Formula/wasm4.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/wasm4.rb:33]: agree, we dont have to rename the artifacts?

Comment (SMillerDev) [reply Formula/wasm4.rb:33]: You can make a symlink. But additionally it's not up to Homebrew to fix upstream issues.

**Comment by chenrui333** [comment]
Also, somehow I cannot build this on my machine

```
/tmp/wasm4-20221230-78678-1798o8k/runtimes/native/vendor/cubeb/src/cubeb_jack.cpp:838:39: warning: result of comparison against a string literal is unspecified (use an explicit string comparison function instead) [-Wstring-compare]
      (output_device && output_device != JACK_DEFAULT_OUT)) {
                                      ^  ~~~~~~~~~~~~~~~~
/tmp/wasm4-20221230-78678-1798o8k/runtimes/native/vendor/cubeb/src/cubeb_jack.cpp:1069:29: error: use of undeclared identifier 'strdup'
    (*device)->input_name = strdup(j_in);
                            ^
/tmp/wasm4-20221230-78678-1798o8k/runtimes/native/vendor/cubeb/src/cubeb_jack.cpp:1070:30: error: use of undeclared identifier 'strdup'
    (*device)->output_name = strdup(j_out);
                             ^
/tmp/wasm4-20221230-78678-1798o8k/runtimes/native/vendor/cubeb/src/cubeb_jack.cpp:1072:29: error: use of undeclared identifier 'strdup'
    (*device)->input_name = strdup(j_in);
                            ^
/tmp/wasm4-20221230-78678-1798o8k/runtimes/native/vendor/cubeb/src/cubeb_jack.cpp:1073:30: error: use of undeclared identifier 'strdup'
    (*device)->output_name = strdup(empty);
                             ^
/tmp/wasm4-20221230-78678-1798o8k/runtimes/native/vendor/cubeb/src/cubeb_jack.cpp:1075:29: error: use of undeclared identifier 'strdup'
    (*device)->input_name = strdup(empty);
                            ^
/tmp/wasm4-20221230-78678-1798o8k/runtimes/native/vendor/cubeb/src/cubeb_jack.cpp:1076:30: error: use of undeclared identifier 'strdup'
    (*device)->output_name = strdup(j_out);
                             ^
2 warnings and 6 errors generated.
```

**Pull request closed without merging**